### ISSUE TITLE
When using a transaction and table valued parameter throws exception

### ISSUE THREAD

**Issue by andymac4182** [jonwagner/Insight.Database#360]
## Type
What kind of issue is this?

```
[x] Bug report.
[ ] Feature request.
```
## Current Behavior

```csharp
Task<IList<int>> MoveHashesFromAvailableToUnused(Guid ticketPoolId, List<VarcharTable> VarcharTable);
```
is defined on an interface that is opened with a transaction `.OpenWithTransactionAsAsync<IMigrateEtixRepository>()`

When trying to pass in a value to varchartable it throws the following exception.
`System.InvalidOperationException: ExecuteScalar requires the command to have a transaction when the connection assigned to the command is in a pending local transaction.  The Transaction property of the command has not been initialized.`

## Expected behavior

Enrol the check in the transaction.

I believe GetTableParameterTypeName needs to pass the transaction on this line. https://github.com/jonwagner/Insight.Database/blob/7f541df8bd87d6703e1876fce06414c930a78fec/Insight.Database.Providers.Default/SqlInsightDbProvider.cs#L445

eg.
```csharp
p.TypeName = tableTypes.Where(t => command.Connection.ExecuteScalarSql<int>("SELECT COUNT(*) FROM sys.table_types WHERE NAME = @name", new { name = t }, transaction: command.Transaction) > 0).First();
```
## Steps to Reproduce

See current behavior 

### Any other comment

Here is the stack trace we are receiving
```
System.AggregateException: One or more errors occurred. ---> System.AggregateException: One or more errors occurred. ---> System.AggregateException: One or more errors occurred. ---> System.InvalidOperationException: ExecuteScalar requires the command to have a transaction when the connection assigned to the command is in a pending local transaction.  The Transaction property of the command has not been initialized.
   at System.Data.SqlClient.SqlCommand.ValidateCommand(String method, Boolean async)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteScalar()
   at Insight.Database.DBConnectionExtensions.<>c__DisplayClass163_0`1.<ExecuteScalar>b__1(IDbCommand _, IDataReader __)
   at Insight.Database.DBConnectionExtensions.ExecuteAndAutoClose[T](IDbConnection connection, Func`2 getCommand, Func`3 translate, CommandBehavior commandBehavior)
   at Insight.Database.DBConnectionExtensions.ExecuteScalar[T](IDbConnection connection, String sql, Object parameters, CommandType commandType, Boolean closeConnection, Nullable`1 commandTimeout, IDbTransaction transaction, Object outputParameters)
   at Insight.Database.SqlInsightDbProvider.<>c__DisplayClass21_0.<GetTableParameterTypeName>b__0(String t)
   at System.Linq.Enumerable.WhereArrayIterator`1.MoveNext()
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at Insight.Database.SqlInsightDbProvider.GetTableParameterTypeName(IDbCommand command, IDataParameter parameter, Type listType)
   at Insight.Database.SqlInsightDbProvider.SetupTableValuedParameter(IDbCommand command, IDataParameter parameter, IEnumerable list, Type listType)
   at Insight.Database.CodeGenerator.DbParameterGenerator.ListParameterHelper.ConvertListParameter(IDataParameter parameter, Object value, IDbCommand command)
   at CreateInputParameters-e14e535d-6875-4ac7-8e06-dff22d35c5a7(IDbCommand , Object )
   at Insight.Database.DBCommandExtensions.AddParameters(IDbCommand cmd, Object parameters)
   at Insight.Database.DBConnectionExtensions.CreateCommand(IDbConnection connection, String sql, Object parameters, CommandType commandType, Nullable`1 commandTimeout, IDbTransaction transaction)
   at Insight.Database.DBConnectionExtensions.<>c__DisplayClass4_0`1.<QueryAsync>b__0(IDbConnection c)
   at Insight.Database.DBConnectionExtensions.<>c__DisplayClass34_0`1.<ExecuteAsyncAndAutoClose>b__0(Task`1 t)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Insight.Database.DBConnectionExtensions.<>c__DisplayClass34_0`1.<ExecuteAsyncAndAutoClose>b__1(Task`1 t)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
```

**Comment by jonwagner** [comment]
Fixed by #361 - will be released in v6.2

**Comment by jonwagner** [comment]
I'm trying to close 3 more issues before I do a build, but my calendar hasn't been cooperating. I think I should be able to get them done by next week.
